### PR TITLE
Dkg generations

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_node/src/node/core/split_barrier.rs [hunk 1]
```diff
@@ -31,29 +33,35 @@ impl SplitBarrier {
...
     #[instrument(skip(self), level = "trace", name = "split barrier processing")]
-    pub(crate) fn process(
+    pub(crate) async fn process(
         &mut self,
         our_prefix: &Prefix,
         section_auth: SectionAuth<SectionAuthorityProvider>,
         keyed_sig: KeyedSig,
+        generation: Generation,
     ) -> Vec<Entry> {
+        debug!(">>>>Processing split barrier for gen: {generation}");
```
Comment (Yoga07) [inline sn_node/src/node/core/split_barrier.rs]: Can be cleaned up?

File: sn_node/src/node/core/split_barrier.rs [hunk 2]
```diff
@@ -31,29 +33,35 @@ impl SplitBarrier {
...
-            mem::take(&mut self.0)
+        let (mut give, mut keep): (Vec<Entry>, Vec<Entry>) =
+            gen_entries
                 .into_iter()
                 .partition(|(cached_section_auth, _)| {
                     cached_section_auth.prefix() == section_auth.prefix().sibling()
                 });
-        self.0 = keep;
 
+        // let _prev = gen_map_guard.insert(generation, keep.clone());
```
Comment (Yoga07) [inline sn_node/src/node/core/split_barrier.rs:60]: Can be cleaned up?

File: sn_node/src/node/core/split_barrier.rs [hunk 3]
```diff
@@ -31,29 +33,35 @@ impl SplitBarrier {
...
         if give.is_empty() {
             // No sibling found. Cache this update until we see the sibling update.
-            self.0.push((section_auth, keyed_sig));
+            keep.push((section_auth, keyed_sig));
```
Comment (Yoga07) [inline sn_node/src/node/core/split_barrier.rs:63]: Could there be a scenario where give is some and keep is empty? Just out of curiosity.

Comment (joshuef) [reply sn_node/src/node/core/split_barrier.rs:63]: I dont think so. If we have one already, we get give (if its new). I'm not 100% here though, mostly just following the prior logic

**Pull request merged**